kubernetes/linux/main.sh

```diff
@@ -1,5 +1,44 @@
 #!/bin/bash
 
+# check to see if Ikey is properly formatted an instrumentation key
+if [[ $APPLICATIONINSIGHTS_AUTH =~ ^[A-Za-z0-9=]+$ ]]  # regex for base64 encoded data
+then
+      # base64 encoded ikey was passed, decode ikey and write to environment variable
+      export APPLICATIONINSIGHTS_AUTH=$(echo $APPLICATIONINSIGHTS_AUTH )
+else
+      # need to fetch ikey from storage account
+      for BACKOFF in {0..5}; do
+            KEY=$(curl -sS $APPLICATIONINSIGHTS_AUTH )
```

Comment:
In hybrid environments, customer can be configured to proxy in that case, this has to be invoked with proxy and if proxy set similar to other external endpoints we do here in this file. Though JEDI not supported yet for Azure Arc, but it will be good to cover similar to other external endpoints we do in the code.

Reply:
Discussed with Arc team and @joby-thomas was also there in the meeting. Eventually, they want to enable in these clouds but not in near term. probable  you can defer this and we can enable it when we can make this change when we add support.

Reply:
Cool, we'll defer it then